client.go

```diff
@@ -97,6 +100,7 @@ type client struct {
 	seedBrokers []*Broker
 	deadSeeds   []*Broker
 
+	controller   *Broker                                 // cluster controller broker
```

Comment:
As with the coordinator map, I'd prefer to store the int32 id here instead of the broker itself. It's easier to reason about object ownership and garbage-collection that way.